### PR TITLE
Migrate `ContentSelector` to TypeScript

### PR DESCRIPTION
**Pull request by acelaya**
This is part of https://github.com/hypothesis/lms/issues/5013

### CODE REVIEW

File: lms/static/scripts/frontend_apps/components/ContentSelector.tsx [hunk 1]
```diff
@@ -1,45 +1,44 @@
 import { FullScreenSpinner, LabeledButton } from '@hypothesis/frontend-shared';
 import { useMemo, useState } from 'preact/hooks';
 
+import type { Book, File, Chapter } from '../api-types';
 import { useConfig } from '../config';
 import { PickerCanceledError } from '../errors';
+import type { Content } from '../utils/content-item';
 import { GooglePickerClient } from '../utils/google-picker-client';
 import { OneDrivePickerClient } from '../utils/onedrive-picker-client';
+import type { ErrorInfo } from './FilePickerApp';
```
Comment (lyzadanger) [inline lms/static/scripts/frontend_apps/components/ContentSelector.tsx:10]: This import should probably be grouped with the next set of imports (it is alphabetically after `BookPicker`).

File: lms/static/scripts/frontend_apps/components/ContentSelector.tsx [hunk 2]
```diff
@@ -1,45 +1,44 @@
...
- * @typedef ContentSelectorProps
- * @prop {DialogType} [defaultActiveDialog] - Used for testing
- * @prop {(ei: ErrorInfo) => void} onError
- * @prop {(c: Content) => void} onSelectContent
- */
+type ContentSelectorProps = {
+  onError: (ei: ErrorInfo) => void;
+  onSelectContent: (c: Content) => void;
+
+  /** Used for testing */
```
Comment (lyzadanger) [inline lms/static/scripts/frontend_apps/components/ContentSelector.tsx:30]: Line break above this comment not required, but not a big deal to leave it, either.

File: lms/static/scripts/frontend_apps/components/ContentSelector.tsx [hunk 3]
```diff
@@ -1,45 +1,44 @@
...
+  | 'vitalSourceBook'
+  | null;
 
-/**
- * @typedef ContentSelectorProps
- * @prop {DialogType} [defaultActiveDialog] - Used for testing
- * @prop {(ei: ErrorInfo) => void} onError
- * @prop {(c: Content) => void} onSelectContent
- */
+type ContentSelectorProps = {
```
Comment (lyzadanger) [inline lms/static/scripts/frontend_apps/components/ContentSelector.tsx:26]: This type should be exported.

**Review by lyzadanger — APPROVED** [review]
The component's props type should be exported. Otherwise this is looking good!

File: lms/static/scripts/frontend_apps/components/ContentSelector.tsx (hunk 2 above)
Comment (acelaya) [reply lms/static/scripts/frontend_apps/components/ContentSelector.tsx:30]: Ok, just for me to understand. Does that mean there should be an empty line only if the comment is multi-line?

File: lms/static/scripts/frontend_apps/components/ContentSelector.tsx (hunk 3 above)
Comment (acelaya) [reply lms/static/scripts/frontend_apps/components/ContentSelector.tsx:26]: Why is it? I don't see it being used outside of this component, but maybe I'm missing something 🤔

File: lms/static/scripts/frontend_apps/components/ContentSelector.tsx (hunk 2 above)
Comment (lyzadanger) [reply lms/static/scripts/frontend_apps/components/ContentSelector.tsx:29]: That's how I've been doing it, but I'm not the boss of everyone :)

File: lms/static/scripts/frontend_apps/components/ContentSelector.tsx (hunk 3 above)
Comment (lyzadanger) [reply lms/static/scripts/frontend_apps/components/ContentSelector.tsx]: It's our convention to export the prop typing for components as they are part of the API of the module, if that makes sense. See other component conversions for examples.

File: lms/static/scripts/frontend_apps/components/ContentSelector.tsx (hunk 2 above)
Comment (acelaya) [reply lms/static/scripts/frontend_apps/components/ContentSelector.tsx:29]: Works for me :)

File: lms/static/scripts/frontend_apps/components/ContentSelector.tsx (hunk 3 above)
Comment (acelaya) [reply lms/static/scripts/frontend_apps/components/ContentSelector.tsx]: Got it.

**Pull request merged**